frontend/src/__tests__/cypress/cypress/utils/modelRegistryUtils.ts

```diff
@@ -10,14 +10,25 @@ import { modelRegistry } from '#~/__tests__/cypress/cypress/pages/modelRegistry'
  * @param timeout Optional timeout for button interactions
  */
 export const clickRegisterModelButton = (timeout?: number): Cypress.Chainable => {
-  return cy.get('body').then(($body) => {
-    if ($body.find('[data-testid="empty-model-registry-primary-action"]').length > 0) {
-      // Registry is empty, use empty state button
-      cy.log('Found empty registry button, clicking it');
-      return modelRegistry.findEmptyRegisterModelButton(timeout).click();
-    }
-    // Registry has models, use regular register button
-    cy.log('Empty registry button not found, using regular register button');
-    return modelRegistry.findRegisterModelButton(timeout).click();
-  });
+  // Wait for either button to appear in DOM
+  return cy
+    .get('body')
+    .should(() => {
+      const body = Cypress.$('body');
+      const hasEmptyButton =
+        body.find('[data-testid="empty-model-registry-primary-action"]').length > 0;
```

Comment:
The `data-testid="empty-model-registry-primary-action"` and `data-testid="register-model-button"` could probably moved to the modelRegistry PO, but it's not critical imo